server/src/main/java/org/opensearch/action/support/nodes/TransportNodesAction.java

```diff
@@ -234,23 +247,18 @@ class AsyncAction {
             this.task = task;
             this.request = request;
             this.listener = listener;
-            if (request.concreteNodes() == null) {
-                resolveRequest(request, clusterService.state());
```

Comment:
i think we cannot remove this block as the derived classes can override this protected method. Ref[1](https://github.com/rajiv-kv/OpenSearch/blob/0040f4b766459610fae3a0342f26d8f78735778e/server/src/test/java/org/opensearch/action/support/nodes/TransportNodesActionTests.java#L347)

Lets continue calling the old method `resolveRequest` and reset after the call based on the include flag.